plugins/base/src/main/kotlin/translators/descriptors/DescriptorAccessorConventionUtil.kt

```diff
@@ -10,29 +10,93 @@ import org.jetbrains.kotlin.load.java.propertyNamesBySetMethodName
 
 internal data class DescriptorFunctionsHolder(
     val regularFunctions: List<FunctionDescriptor>,
-    val accessors: Map<PropertyDescriptor, List<FunctionDescriptor>>
+    val accessors: Map<PropertyDescriptor, DescriptorAccessorHolder>
 )
 
-internal fun splitFunctionsAndAccessors(
+internal data class DescriptorAccessorHolder(
+    val getter: FunctionDescriptor? = null,
+    val setter: FunctionDescriptor? = null
+)
+
+/**
+ * Separate regular Kotlin/Java functions and inherited Java accessors
+ * to properly display properties inherited from Java.
+ *
+ * Take this example:
+ * ```
+ * // java
+ * public class JavaClass {
+ *     private int a = 1;
+ *     public int getA() { return a; }
+ *     public void setA(int a) { this.a = a; }
+ * }
+ *
+ * // kotlin
+ * class Bar : JavaClass() {
+ *     fun foo() {}
+ * }
+ * ```
+ *
+ * It should result in:
+ * - 1 regular function `foo`
+ * - Map a=[`getA`, `setA`]
+ */
+internal fun splitFunctionsAndInheritedAccessors(
     properties: List<PropertyDescriptor>,
     functions: List<FunctionDescriptor>
 ): DescriptorFunctionsHolder {
+    val (javaMethods, kotlinFunctions) = functions.partition { it is JavaMethodDescriptor }
+    if (javaMethods.isEmpty()) {
+        return DescriptorFunctionsHolder(regularFunctions = kotlinFunctions, emptyMap())
+    }
+
     val propertiesByName = properties.associateBy { it.name.asString() }
-    val regularFunctions = mutableListOf<FunctionDescriptor>()
-    val accessors = mutableMapOf<PropertyDescriptor, MutableList<FunctionDescriptor>>()
-    functions.forEach { function ->
+    val regularFunctions = ArrayList<FunctionDescriptor>(kotlinFunctions)
+
+    val accessors = mutableMapOf<PropertyDescriptor, DescriptorAccessorHolder>()
+    javaMethods.forEach { function ->
         val possiblePropertyNamesForFunction = function.toPossiblePropertyNames()
         val property = possiblePropertyNamesForFunction.firstNotNullOfOrNull { propertiesByName[it] }
         if (property != null && function.isAccessorFor(property)) {
-            accessors.getOrPut(property, ::mutableListOf).add(function)
+            accessors.compute(property) { prop, accessorHolder ->
+                if (function.isGetterFor(prop))
+                    accessorHolder?.copy(getter = function) ?: DescriptorAccessorHolder(getter = function)
+                else
+                    accessorHolder?.copy(setter = function) ?: DescriptorAccessorHolder(setter = function)
+            }
         } else {
             regularFunctions.add(function)
         }
     }
+
+    val accessorLookalikes = removeNonAccessorsReturning(accessors)
```

Comment:
I would make a filter here. It is more obvious

Reply:
Unfortunately, one simple filter would not do :( We need to both remove elements from `accessors` and add removed elements to `regularFunctions` (so it involves mapping of elements also)

Filter would only return the elements which need to be added to `regularFunctions`, but it will not remove it from `accessors`. 

This could be done in multiple steps, something like this, but IMHO it involves many excessive operations which can be substituted with a single iteration
```kotlin
val nonAccessibleProperties: Map<PropertyDescriptor, DescriptorAccessorHolder> =
        accessors.filterValues { it.getter == null && it.setter != null }

val functionsOfNonAccessibleProperties = nonAccessibleProperties.values.map { it.setter!! }
regularFunctions.addAll(functionsOfNonAccessibleProperties)

nonAccessibleProperties.keys.forEach { accessors.remove(it) }
```